src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMockFactory.java

```diff
@@ -1,67 +1,45 @@
 package be.seeseemelk.mockbukkit.entity;
 
-import java.util.HashSet;
-import java.util.Random;
-import java.util.Set;
-import java.util.UUID;
-
 import be.seeseemelk.mockbukkit.ServerMock;
+import org.jetbrains.annotations.NotNull;
 
-public final class PlayerMockFactory
-{
-	private static final String[] FIRST_NAMES = { "James", "Mary", "John", "Particia", "Robert", "Jennifer", "Michael", "Elizabeth", "William", "Linda" };
-	private static final String[] LAST_NAMES = { "Smith", "Johnson", "Jones", "Brown", "Davis", "Miller", "Wilson", "Moore", "Taylor", "Anderson" };
+import java.util.Random;
+import java.util.UUID;
 
+public final class PlayerMockFactory {
 	private final ServerMock server;
-	private Random random = new Random();
-	private Set<String> usedNames = new HashSet<>();
+	private final Random random = new Random();
+	private int currentNameIndex;
 
-	public PlayerMockFactory(ServerMock server)
+	public PlayerMockFactory(@NotNull ServerMock server)
 	{
+		this.currentNameIndex = 0;
 		this.server = server;
 	}
 
-	/**
-	 * Generates a random name.
-	 *
-	 * @return A randomly generated name.
-	 */
-	private String getRandomName()
-	{
-		String firstName = FIRST_NAMES[random.nextInt(FIRST_NAMES.length)];
-		String lastName = LAST_NAMES[random.nextInt(LAST_NAMES.length)];
-		return firstName + " " + lastName;
-	}
-
 	/**
 	 * Gets a unique random name. Any name that is generated will never be generated again.
 	 *
 	 * @return A unique random name.
 	 */
-	private String getUniqueRandomName()
+	private @NotNull String getUniqueRandomName()
```

Comment:
```suggestion
	private @NotNull String getUniqueName()
```
It's not random anymore